apps/server/src/utils/ElasticSearchClient.ts

```diff
@@ -0,0 +1,114 @@
+import type { estypes } from '@elastic/elasticsearch'
+import { Client } from '@elastic/elasticsearch'
+import { readFileSync } from 'fs'
+import type { LocalInstitution } from 'src/shared/contract'
+
+export default class ElasticsearchClient {
+  private readonly client: Client
+  private static _instance: ElasticsearchClient
+
+  private constructor () {
+    this.client = new Client({ node: 'http://localhost:9200' })
+  }
+
+  static async initialize () {
+    const client = this.getInstance().client
+
+    const elasticSearchLoaded = await client.indices.exists({ index: 'institutions' })
+    if (!elasticSearchLoaded) {
+      await this.reIndexElasticSearch()
+    }
+  }
+
+  static getInstance () {
+    if (this._instance) {
+      return this._instance
+    }
+
+    this._instance = new ElasticsearchClient()
+    return this._instance
+  }
+
+  static async reIndexElasticSearch () {
+    const client = this.getInstance().client
+    try {
+      await client.indices.delete({
+        index: 'institutions'
+      })
+    } catch {
+      console.log('Elasticsearch "institutions" index did not exist')
+    }
+    await client.indices.create({ index: 'institutions' })
+    console.log('Elasticsearch indexing institutions')
+    const dataFilePath = 'ucw_institution_mapping.json'
+    const rawData = readFileSync(dataFilePath)
```

Comment:
I think we should build this so it is more easily added to when we have an institutions api we are polling.

On initial load it should pull from the json file and load it into the cache. After initialization we should only be looking in the cache.

Reply:
Should we save this work for when we have an institutions api to get data from?